### PR TITLE
Recover in-progress timer sessions after restart

### LINKED ISSUE #136: Session recovery for in-progress sessions
## Goal
Persist and restore in-progress timer state after app restart/crash.

## Scope
- Persist active phase, remaining seconds, running/paused status
- Persist active task label and active profile
- Restore valid in-progress state on startup
- Clear recovery state when session naturally ends or resets

## Acceptance criteria
- Restarting during active focus/break restores expected timer state
- Invalid/stale recovery state is safely ignored with clear user feedback
- Existing config/stats persistence behavior remains unchanged

### PR DESCRIPTION
**Pull request by utilForever**
## What

Add in-progress session recovery for the TUI timer lifecycle.

- Introduce `src/session_recovery.rs` to persist and load a typed snapshot (`phase`, `status`, `remaining_secs`, selected task label, selected profile).
- Restore valid running/paused focus or break state on app startup and surface a recovery notice in the timer view.
- Ignore and clear invalid or stale recovery snapshots with explicit user-facing warning text.
- Keep recovery state synchronized during timer transitions and clear it when the session is no longer recoverable.
- Add recovery-focused tests in `app.rs` and in the new recovery module.
- Document the feature in `README.md`.

## Why

Closes #136.

Restarting during an active session currently loses in-progress timer context. This change keeps focus/break progress and session context recoverable after restart/crash while failing safely for bad persisted state. Recovery resumes from the last saved remaining time (no downtime subtraction), which matches the issue decision for predictable behavior.

### Required

- [x] `cargo check --all` passes
- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes
- [x] `cargo test --all` passes
- [x] I linked the related issue (for example: `Closes #25`)

### Functional Validation

- [x] Pomodoro timer behavior was verified for this change (if applicable)
- [x] Site blocking behavior was verified for this change (if applicable)
- [x] WakaTime integration behavior was verified for this change (if applicable)
- [x] I added or updated tests for changed behavior (if applicable)

### Configuration & Docs

- [x] User-facing docs were updated (`README.md` or relevant docs, if applicable)
- [x] New dependencies/configuration are documented (if applicable)
- [x] No sensitive values or credentials were introduced

### If Applicable

- [x] Security impact considered (run `cargo audit` locally if needed)
- [x] Breaking behavior changes are clearly described in this PR

## Summary by CodeRabbit

* **New Features**
  * Timer state is persisted during running/paused focus or break phases and automatically restored on app restart/crash (phase, remaining time, task label, profile). Persistence is kept up-to-date during runtime and when planner/label/profile changes occur.

* **Bug Fixes**
  * Invalid or stale recovery data is ignored with a warning; recovery data is cleared when a phase ends or the user resets/skips. Startup load errors surface a clear notification.

* **Documentation**
  * Added "Session recovery" docs explaining behavior and visible notifications.

* **Tests**
  * Added unit tests for restore, invalid-snapshot handling, runtime save/clear, and label-change behavior.

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -1021,6 +1026,113 @@ impl App {
...
+        self.apply_blocking_for_phase();
+        self.sync_task_planner_state();
+        Ok(())
+    }
+
+    fn sync_recovery_snapshot(&mut self) {
+        let snapshot = InProgressSessionSnapshot::from_timer_state(
+            &self.timer,
+            self.selected_task_label.clone(),
+            self.selected_profile,
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs:1123]: Fixed in 4f57dee.

- sync_recovery_snapshot now persists ctive_focus_task_label while a focus session is active, with a fallback to the selected label when needed.
- Added regression test 
ecovery_snapshot_prefers_active_focus_label_over_selected_label to cover mid-focus planner label changes.

Thanks for the catch.

File: src/app.rs [hunk 2]
```diff
@@ -1021,6 +1026,121 @@ impl App {
...
+                    self.config_error = Some(format!("session recovery save failed: {error}"));
+                }
+            }
+            None => {
+                if let Err(error) = session_recovery::clear() {
+                    self.config_error = Some(format!("session recovery clear failed: {error}"));
+                }
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs:1138]: Addressed in 2305cb7.

- Added sync_recovery_snapshot() immediately after planner label mutations in both commit_planner_input and select_planner_label (right after sync_task_planner_state()).
- Added regression test planner_label_change_during_running_break_updates_recovery_snapshot to verify label changes during a running break are persisted to recovery immediately.

Thanks for highlighting this edge case.

File: src/app.rs [hunk 3]
```diff
@@ -1021,6 +1026,121 @@ impl App {
...
+            Ok(snapshot) => snapshot,
+            Err(error) => {
+                self.phase_notification =
+                    Some(format!("Ignored saved in-progress session: {error}."));
+                if let Err(clear_error) = session_recovery::clear() {
+                    self.config_error = Some(format!(
+                        "session recovery cleanup failed after load error: {clear_error}"
+                    ));
+                }
+                return;
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs:1035]: Addressed in 536caa1.

I removed the session_recovery::clear() call from the generic load() error path so transient read/permission errors no longer delete potentially valid recovery data. We still clear recovery state for validated invalid snapshots during apply/validation failure.

File: src/app.rs [hunk 4]
```diff
@@ -1021,6 +1026,121 @@ impl App {
...
+        let mut recovered_timer = TimerState::with_profile(
+            profile_spec.focus_secs,
+            profile_spec.short_break_secs,
+            profile_spec.long_break_secs,
+            profile_spec.long_break_interval,
+        );
+        recovered_timer.phase = snapshot.phase();
+        recovered_timer.status = snapshot.status();
+        recovered_timer.remaining_secs = snapshot.remaining_secs;
+        snapshot.validate_for_timer(&recovered_timer)?;
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs:1075]: Addressed in 536caa1.

I added pomodoros_completed to InProgressSessionSnapshot (with #[serde(default)] for compatibility), persist it when saving recovery state, and restore it into TimerState before validation. I also added startup_restores_pomodoro_count_for_phase_cadence to verify cadence is preserved after recovery.

File: src/app.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs:1035]: Addressed in 536caa1.

I removed the `session_recovery::clear()` call from the generic `load()` error path so transient read or permission errors no longer delete potentially valid recovery data. We still clear recovery state for validated invalid snapshots during apply/validation failure.

File: src/app.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (utilForever) [reply src/app.rs:1075]: Addressed in 536caa1.

I added `pomodoros_completed` to `InProgressSessionSnapshot` (with `#[serde(default)]` for backward compatibility), persist it when saving recovery state, and restore it into `TimerState` before validation. I also added `startup_restores_pomodoro_count_for_phase_cadence` to verify cadence remains correct after recovery.
